client/js/drawModule.js

```diff
@@ -0,0 +1,1031 @@
+      diners.textContent = reservationsResponse[j].partySize;
+      row.appendChild(diners);
+
+      tableBody.appendChild(row);
+    }
+
+    reservationsTable.appendChild(tableBody);
+    return reservationsTable;
+  }
+
+
+
+  function addCreateReviewLink(restaurantName) {
+    var userInfo = JSON.parse(localStorage.getItem('userInfo'));
+
+    if (! connectionsAPI.hasRole(userInfo,
+      connectionsAPI.rol.endUser)) {
+      return null;
+    }
+
+    var createReviewLink = document.createElement('A');
+    createReviewLink.textContent = 'Create review';
+    createReviewLink.onclick = (function(restaurantName) {
+      return function() {
+      createAndShowReviewForm(restaurantName);
+      };
+    })(restaurantName);
+
+    return createReviewLink;
+  }
+
+
+  function createAndShowReviewForm(restaurantName) {
+    var reviewForm = createReviewForm(restaurantName);
+    var title = 'Create review for: ' + restaurantName;
+    setPopupTitle(title);
+    setPopupContent(reviewForm);
+    openPopUpWindow();
+  }
+
+  //create a form for creating or updating a review
+  function createReviewForm(restaurantName, review) {
+    var reviewForm = document.createElement('FORM');
+    reviewForm.name = 'editReviewForm';
+    reviewForm.className = 'editReviewForm';
+
+    reviewForm.onsubmit = function() {
+      var ratingValue =
+        parseInt(document.forms.editReviewForm.ratingValue.value);
+
+      var reviewBody = document.forms.editReviewForm.reviewBody.value;
+      if (review) {
+        updateReviewAction(review.name, ratingValue, reviewBody);
+      }
+      else {
+        createNewReviewAction(restaurantName, ratingValue, reviewBody);
+      }
+
+      return false;
+    };
+
+    var reviewLabel = document.createElement('LABEL');
+    reviewLabel.textContent = 'Your review: ';
+    reviewForm.appendChild(reviewLabel);
+    reviewForm.appendChild(document.createElement('BR'));
+
+    var reviewBody = document.createElement('TEXTAREA');
+    reviewBody.name = 'reviewBody';
+    reviewForm.appendChild(reviewBody);
+    reviewForm.appendChild(document.createElement('BR'));
+
+    var ratingLabel = document.createElement('LABEL');
+    ratingLabel.textContent = 'Rating value: ';
+    reviewForm.appendChild(ratingLabel);
+
+    var ratingValueSelect = document.createElement('SELECT');
+    ratingValueSelect.name = 'ratingValue';
+
+    var option;
+    for (var i = 0; i <= maxRating; i++) {
+      option = document.createElement('OPTION');
+      option.value = i;
+      option.textContent = i + ' Star' + (1 != i ? 's' : '');
+      ratingValueSelect.appendChild(option);
+    }
+
+    reviewForm.appendChild(ratingValueSelect);
+
+    var submit = document.createElement('INPUT');
+    submit.type = 'submit';
+    submit.value = 'Create Review';
+    submit.name = 'submitReview';
+    reviewForm.appendChild(submit);
+
+
+    return reviewForm;
+  }
+
+  //inicialize a review form when update a review
+  //instead of create a new one
```

Comment:
"//inicialize a review form when update a review
//instead of create a new one" ->
"//inicialize a review form when updating a review
//instead of creating a new one"